src/openrct2/world/MapAnimation.cpp

```diff
@@ -222,446 +117,238 @@ static bool MapAnimationInvalidateSmallScenery(const CoordsXYZ& loc)
     return true;
 }
 
-/**
- *
- *  rct2: 0x006BA2BB
- */
-static bool MapAnimationInvalidateBanner(const CoordsXYZ& loc)
+static bool UpdateTileAnimations(const TileCoordsXY& coords)
 {
-    TileCoordsXYZ tileLoc{ loc };
-    TileElement* tileElement;
-
-    tileElement = MapGetFirstElementAt(loc);
+    bool wasUpdated = false;
+    auto tileElement = MapGetFirstElementAt(coords);
     if (tileElement == nullptr)
-        return true;
-    do
     {
-        if (tileElement->BaseHeight != tileLoc.z)
-            continue;
-        if (tileElement->GetType() != TileElementType::Banner)
-            continue;
-        MapInvalidateTileZoom1({ loc, loc.z, loc.z + 16 });
         return false;
-    } while (!(tileElement++)->IsLastForTile());
-
-    return true;
-}
-
-/**
- *
- *  rct2: 0x006B94EB
- */
-static bool MapAnimationInvalidateLargeScenery(const CoordsXYZ& loc)
-{
-    TileCoordsXYZ tileLoc{ loc };
-    TileElement* tileElement;
+    }
 
-    bool wasInvalidated = false;
-    tileElement = MapGetFirstElementAt(loc);
-    if (tileElement == nullptr)
-        return true;
     do
     {
-        if (tileElement->BaseHeight != tileLoc.z)
-            continue;
-        if (tileElement->GetType() != TileElementType::LargeScenery)
-            continue;
+        auto baseZ = tileElement->GetBaseZ();
+        CoordsXYZ loc{ coords.ToCoordsXY(), baseZ };
 
-        auto* sceneryEntry = tileElement->AsLargeScenery()->GetEntry();
-        if (sceneryEntry != nullptr && sceneryEntry->flags & LARGE_SCENERY_FLAG_ANIMATED)
+        switch (tileElement->GetType())
         {
-            MapInvalidateTileZoom1({ loc, loc.z, loc.z + 16 });
-            wasInvalidated = true;
+            case TileElementType::Entrance:
+                wasUpdated |= UpdateEntranceAnimation(tileElement->AsEntrance(), loc, baseZ);
+                break;
+            case TileElementType::Path:
+                wasUpdated |= UpdatePathAnimation(tileElement->AsPath(), loc, baseZ);
+                break;
+            case TileElementType::SmallScenery:
+                wasUpdated |= UpdateSmallSceneryAnimation(tileElement->AsSmallScenery(), loc, baseZ);
+                break;
+            case TileElementType::Track:
+                wasUpdated |= UpdateTrackAnimation(tileElement->AsTrack(), loc, baseZ);
+                break;
+            case TileElementType::Banner:
+                wasUpdated |= UpdateBannerAnimation(tileElement->AsBanner(), loc, baseZ);
+                break;
+            case TileElementType::LargeScenery:
+                wasUpdated |= UpdateLargeSceneryAnimation(tileElement->AsLargeScenery(), loc, baseZ);
+                break;
+            case TileElementType::Wall:
+                wasUpdated |= UpdateWallAnimation(tileElement->AsWall(), loc, baseZ);
+                break;
+            default:
+                break;
         }
     } while (!(tileElement++)->IsLastForTile());
 
-    return !wasInvalidated;
+    return wasUpdated;
```

Comment:
its not really wasUpdated its an `animationStillRequired`.

Reply:
True, I'll change it.